packages/EasyCore/src/Bridge/Symfony/Validator/Constraints/AlphanumericHyphenValidator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Bridge\Symfony\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class AlphanumericHyphenValidator extends ConstraintValidator
+{
+    public function validate($value, Constraint $constraint): void
+    {
+        if ($constraint instanceof AlphanumericHyphen === false) {
+            throw new UnexpectedTypeException($constraint, AlphanumericHyphen::class);
+        }
+
+        if ($value === null || $value === '') {
+            return;
+        }
+
+        if (\is_scalar($value) === false
+            && (\is_object($value) && \method_exists($value, '__toString')) === false) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $value = (string)$value;
+
+        if (\preg_match('/^[a-z0-9_-]+$/i', $value) === 0) {
```

Comment:
The message in the constraint class says that only alphanumeric and hyphens are allowed, but I see an underscore symbol in the regexp here. Is it okay?

Reply:
I think that in this case, it is acceptable.
@roman-eonx - what do you think?

Reply:
I have nothing against it, but in this case, I think it would be a good idea to extend the message in the constraint class

Reply:
Yes, hyphens = underscores + minuses in this case (as our technical writer decided). And the validation message was updated by him from the old "dashes" one.